src/adapters/bedrock/chat.ts

```diff
@@ -0,0 +1,286 @@
+      return { tokenLimit: 3 * 1024 };
+    } else if (
+      this.modelId.includes("titan-text-express") ||
+      this.modelId.includes("anthropic.claude-v2") ||
+      this.modelId.includes("anthropic.claude-instant-v1") ||
+      this.modelId.includes("anthropic.claude-3-sonnet") ||
+      this.modelId.includes("anthropic.claude-3-haiku") ||
+      this.modelId.includes("anthropic.claude-3-opus") ||
+      this.modelId.includes("meta.llama2") ||
+      this.modelId.includes("cohere.command-text") ||
+      this.modelId.includes("cohere.command-light")
+    ) {
+      return { tokenLimit: 4 * 1024 };
+    } else if (
+      this.modelId.includes("titan-text-lite") ||
+      this.modelId.includes("anthropic.claude-3-5-sonnet") ||
+      this.modelId.includes("anthropic.claude-3-5-haiku") ||
+      this.modelId.includes("meta.llama3-8b") ||
+      this.modelId.includes("meta.llama3-70b") ||
+      this.modelId.includes("ai21.j2")
+    ) {
+      return { tokenLimit: 8 * 1024 };
+    } else if (
+      this.modelId.includes("mistral.mistral-7b") ||
+      this.modelId.includes("mistral.mixtral-8x7b") ||
+      this.modelId.includes("mistral.mistral-small")
+    ) {
+      return { tokenLimit: 32 * 1024 };
+    } else if (
+      this.modelId.includes("meta.llama3-1") ||
+      this.modelId.includes("meta.llama3-2") ||
+      this.modelId.includes("mistral.mistral-large") ||
+      this.modelId.includes("cohere.command-r")
+    ) {
+      return { tokenLimit: 128 * 1024 };
+    } else if (this.modelId.includes("ai21.jamba")) {
+      return { tokenLimit: 256 * 1024 };
+    }
+
+    return {
+      tokenLimit: Infinity,
+    };
+  }
+
+  async tokenize(input: BaseMessage[]): Promise<BaseLLMTokenizeOutput> {
+    const contentLength = input.reduce((acc, msg) => acc + msg.text.length, 0);
+    return { tokensCount: Math.ceil(contentLength / 4) };
+  }
+
+  protected convertToConverseMessages(messages: BaseMessage[]): {
+    converseMessages: BedrockMessage[];
+    converseSystem: BedrockSystemContentBlock[];
+  } {
+    const converseSystem: BedrockSystemContentBlock[] = messages
+      .filter((msg) => msg.role === Role.SYSTEM)
+      .map((msg) => {
+        return { text: msg.text };
+      });
+    const converseMessages: BedrockMessage[] = messages
+      .filter((msg) => msg.role !== Role.SYSTEM)
+      .map((msg) => {
+        return {
+          role: msg.role === Role.USER ? Role.USER : Role.ASSISTANT,
+          content: [{ text: msg.text }],
+        };
+      });
+
+    const combinedConverseMessages = converseMessages.reduce<BedrockMessage[]>((acc, curr) => {
+      const lastMessage = acc[acc.length - 1];
+      if (lastMessage && lastMessage.role === Role.USER) {
+        lastMessage.content = lastMessage.content!.concat(curr.content!);
+      } else {
+        acc.push(curr);
+      }
+
+      return acc;
+    }, []);
+
+    return { converseMessages: combinedConverseMessages, converseSystem };
+  }
```

Comment:
I suggest using a similar approach that has been used in the LangChain adapter. It should help with maintainability.

See: https://github.com/i-am-bee/bee-agent-framework/blob/main/src/adapters/langchain/llms/chat.ts#L128-L166

Reply:
And please try to use better names for variables.

Reply:
Bedrock converse and converseStream require not only mapping message roles but also splitting system messages and assistant/user messages into different parameters. I find it hard to create a mapper to achieve the desired input. However, I'm changing the variable names to something more meaningful.